tests/integration/high_availability/conftest.py

```diff
@@ -33,8 +34,10 @@ async def continuous_writes(ops_test: OpsTest) -> None:
 
     yield
 
-    logger.info("Clearing continuous writes")
-    await juju_.run_action(application_unit, "clear-continuous-writes")
+    for attempt in Retrying(stop=stop_after_attempt(10), wait=wait_fixed(30)):
+        with attempt:
+            logger.info("Clearing continuous writes")
+            await juju_.run_action(application_unit, "clear-continuous-writes")
```

Comment:
Got any insight why this fails on the test app?

Reply:
There are a few reasons (like [this](https://github.com/canonical/mysql-k8s-operator/actions/runs/10625153069/job/29454851141#step:37:1037) where a relation user does not exist for the test app to clear continuous writes and [this](https://github.com/canonical/mysql-k8s-operator/actions/runs/10633935256/job/29480518847#step:37:1077) where the database is in `--read-only` mode)

I am starting to lean toward's Alex's caution - may be we shouldnt retry clearing continuous writes and instead focus on the root cause which leads to the relation user not being available or that which leads to the database being in read-only mode

Reply:
In any case, it seems more like a test app issue.
For that we need to either treat error or fail when it should there, validating test-app status in test.
imo, we should not block this though